shared/ssl.sh

```diff
@@ -3,28 +3,37 @@
 # Get a signed certificate for this host.
 #
 # Uses OpenSSL and curl directly to generate a new CSR and get it signed by the
-# Puppet Server CA. Does not support custom CSR extensions at the moment.
+# Puppet Server CA.
 #
 # Intended to be used in place of a full-blown puppet agent run that is solely
 # for getting SSL certificates onto the host.
 #
 # Files will be placed in the same default directory location and structure
 # that the puppet agent would put them, which is /etc/puppetlabs/puppet/ssl.
 #
-# Fails under the following conditions:
-# - CA host cannot be reached
-# - CA already has signed certificate for this host
-# - SSL files already exist on this host
+# The certname can be provided as the first argument to this script, or
+# as the CERTNAME environment variable. If both are found, the argument
+# takes precedence over the environment variable. If neither are found,
+# the HOSTNAME will be used.
+#
+# Supports DNS alt names via the DNS_ALT_NAMES environment variable, which
+# is a comma-separated string of names. The Puppet Server CA must be configured
+# to allow subject alt names, by default it will reject certificate requests
+# with them.
 #
 # Arguments:
-#   $1  (Optional) Certname to use, defaults to $HOSTNAME
+#   $1  (Optional) Certname to use. Overrides the CERTNAME environment variable.
+#                  If neither are set, the $HOSTNAME will be used.
 #
 # Optional environment variables:
+#   CERTNAME               Certname to use, unless an argument is passed in
```

Comment:
I also added this `CERTNAME` variable, which we're passing in as the argument like `/ssl. "$CERTNAME"` in our PE repos, so that'll streamline that a bit.